src/emitter.ts

```diff
@@ -0,0 +1,1126 @@
+                printer.printLineToStack(content);
+            }
+            else {
+                printer.printLine(content);
+            }
+        }
+
+        emitComments(m, printLine);
+
+        switch (m.name) {
+            case "createElement": return emitCreateElementOverloads(m);
+            case "createEvent": return emitCreateEventOverloads(m);
+            case "getElementsByTagName": return emitGetElementsByTagNameOverloads(m);
+            case "querySelector": return emitQuerySelectorOverloads(m);
+            case "querySelectorAll": return emitQuerySelectorAllOverloads(m);
+        }
+        emitSignatures(m, prefix, m.name, printLine);
+    }
+
+    function emitSignature(s: Browser.Signature, prefix: string | undefined, name: string | undefined, printLine: (s: string) => void) {
+        const paramsString = s.param ? paramsToString(s.param) : "";
+        let returnType = convertDomTypeToTsType(s);
+        returnType = s.nullable ? makeNullable(returnType) : returnType;
+        printLine(`${prefix || ""}${name || ""}(${paramsString}): ${returnType};`);
+    }
+
+    function emitSignatures(method: { signature?: Browser.Signature[], "override-signatures"?: string[], "additional-signatures"?: string[] }, prefix: string, name: string, printLine: (s: string) => void) {
+        if (method["override-signatures"]) {
+            method["override-signatures"]!.forEach(s => printLine(`${prefix}${s};`));
+        }
+        else if (method.signature) {
+            if (method["additional-signatures"]) {
+                method["additional-signatures"]!.forEach(s => printLine(`${prefix}${s};`));
+            }
+            method.signature.forEach(sig => emitSignature(sig, prefix, name, printLine));
+        }
+    }
+
+    function emitMethods(prefix: string, emitScope: EmitScope, i: Browser.Interface, conflictedMembers: Set<string>) {
+        // If prefix is not empty, then this is the global declare function addEventListener, we want to override this
+        // Otherwise, this is EventTarget.addEventListener, we want to keep that.
+        function mFilter(m: Browser.Method) {
+            return matchScope(emitScope, m) &&
+                !(prefix !== "" && (m.name === "addEventListener" || m.name === "removeEventListener"));
+        }
+
+        if (i.methods) {
+            mapToArray(i.methods.method)
+                .filter(mFilter)
+                .sort(compareName)
+                .forEach(m => emitMethod(prefix, i, m, conflictedMembers));
+        }
+
+        // The window interface inherited some methods from "Object",
+        // which need to explicitly exposed
+        if (i.name === "Window" && prefix === "declare function ") {
+            printer.printLine("declare function toString(): string;");
+        }
+    }
+
+    /// Emit the properties and methods of a given interface
+    function emitMembers(prefix: string, emitScope: EmitScope, i: Browser.Interface) {
```

Comment:
Would move `prefix` to the end and make it optional.
Also, the only `prefix` value currently (besides `""`) is `"declare var"`, so maybe the parameter should be just `inNamespace = false`.

Reply:
yeah. i need to do this for all prefixes.. but will do this separately. no need to pass a string in.